app/src/main/java/korablique/recipecalculator/database/UserParametersWorker.java

```diff
@@ -44,14 +42,39 @@ public void initCache() {
         return cachedUserParameters;
     }
 
+    private Single<Optional<UserParameters>> requestCurrentUserParametersObservable() {
+        return requestUserParametersByFunction(() -> {
+            AppDatabase database = databaseHolder.getDatabase();
+            UserParametersDao userDao = database.userParametersDao();
+            return userDao.loadCurrentUserParameters();
+        });
+    }
+
+    public Single<Optional<UserParameters>> requestFirstUserParameters() {
+        return requestUserParametersByFunction(() -> {
+            AppDatabase database = databaseHolder.getDatabase();
+            UserParametersDao userDao = database.userParametersDao();
+            return userDao.loadFirstUserParameters();
+        });
+    }
+
     /**
-     * Преобразует вызов {@link #requestCurrentUserParametersImpl}, который должен быть
-     * сделан на потоке БД, в создание Observable.
+     * Makes actual request to database, must be called on database thread.
```

Comment:
Hello.
This comment does not reflect current state of things - the method can be called on any thread - it delegates the work to a background thread by the call:
```
 result.subscribeOn(databaseThreadExecutor.asScheduler())
```
Because the database scheduler is used.

My name is Boris.